.github/workflows/deploy.yaml

```diff
@@ -60,11 +63,13 @@ jobs:
             --baseURL "https://${{ env.CANONICAL_DOMAIN }}/"
 
       - name: Upload artifact
+        if: github.ref == 'refs/heads/main' && github.event_name != 'pull_request'
```

Comment:
nit: I'm pretty sure the latter check is redundant (I believe refs for PRs should follow `refs/pull/<pr_number>/merge`)

Reply:
Good catch but that entire condition will be gone with the dedicated pr workflow